app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt

```diff
@@ -202,95 +221,108 @@ private fun assertHomeToolbar() = onView(ViewMatchers.withResourceName("toolbar"
     .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertOpenTabsHeader() =
-    onView(CoreMatchers.allOf(ViewMatchers.withText("Open tabs")))
+    onView(CoreMatchers.allOf(withText("Open tabs")))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertAddTabButton() =
-    onView(CoreMatchers.allOf(ViewMatchers.withId(R.id.add_tab_button), isDisplayed()))
+    onView(CoreMatchers.allOf(withId(R.id.add_tab_button), isDisplayed()))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertNoTabsOpenedHeader() =
-    onView(CoreMatchers.allOf(ViewMatchers.withText("No open tabs")))
+    onView(CoreMatchers.allOf(withText("No tabs opened")))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertNoTabsOpenedText() {
-    onView(CoreMatchers.allOf(ViewMatchers.withText("Your open tabs will be shown here.")))
+    onView(CoreMatchers.allOf(withText("Your open tabs will be shown here.")))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 }
 
 private fun assertCollectionsHeader() =
-    onView(CoreMatchers.allOf(ViewMatchers.withText("Collections")))
+    onView(CoreMatchers.allOf(withText("Collections")))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertNoCollectionsHeader() =
-    onView(CoreMatchers.allOf(ViewMatchers.withText("No collections")))
+    onView(CoreMatchers.allOf(withText("No collections")))
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertNoCollectionsText() =
     onView(
         CoreMatchers.allOf(
-            ViewMatchers
-                .withText("Collect the things that matter to you. To start, save open tabs to a new collection.")
+            withText("Collect the things that matter to you. To start, save open tabs to a new collection.")
         )
     )
         .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
 private fun assertHomeComponent() = onView(ViewMatchers.withResourceName("home_component"))
     .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
 
-private fun threeDotButton() = onView(allOf(ViewMatchers.withId(R.id.menuButton)))
+private fun threeDotButton() = onView(allOf(withId(R.id.menuButton)))
+
+private fun verifySearchEngineIcon(searchEngineIcon: Bitmap, searchEngineName: String) {
+    onView(withId(R.id.search_engine_icon))
+        .check(matches(withBitmapDrawable(searchEngineIcon, searchEngineName)))
+}
+
+private fun getSearchEngine(searchEngineName: String) =
+    Search(appContext).searchEngineManager.getDefaultSearchEngine(appContext, searchEngineName)
+
+private fun verifyDDGSearchEngineIcon() {
```

Comment:
Yeah again not the biggest fan of these super specific per search engine checks. Can they be re-written to be more generic incase the engines change in product?